### PR TITLE
1letter/fix#32

### PR DESCRIPTION
**Pull request by 1letter**
- this PR change the schema of the portlet
- all boolean fields are now per default not required, because unselection ist with the `required=True` Option not possible
- some infrastructure work for test setup
- reformatting of some files with black and isort
- reformatting of template with zpretty

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -42,7 +42,8 @@
     extras_require={
         "test": [
             "plone.app.testing",
-            "plone.app.contenttypes"
+            "plone.app.contenttypes",
+            "plone.app.robotframework[debug]",
```
Comment (mauritsvanrees) [inline setup.py:46]: There are no robot tests, so this should not be added.

File: base.cfg [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+output = ${buildout:directory}/bin/test-coverage
+mode = 755
+
+
+[createcoverage]
+recipe = zc.recipe.egg
+eggs = createcoverage
+
+
+[robot]
```
Comment (mauritsvanrees) [inline base.cfg:86]: There are no robot tests, so we do not need this.

File: base.cfg [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+
+[i18ndude]
+recipe = zc.recipe.egg
+eggs = i18ndude
+
+[plone-helper-scripts]
+recipe = zc.recipe.egg
+eggs =
+   Products.CMFPlone
```
Comment (mauritsvanrees) [inline base.cfg:105]: instance eggs is enough, without adding CMFPlone again

File: base.cfg [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+parts =
+    instance
+    test
+# we use tox for testing and linting, by default
+#    code-analysis
+    coverage
+    test-coverage
+    createcoverage
+    releaser
```
Comment (mauritsvanrees) [inline base.cfg:14]: We don't need releaser in here. When releasing I use the fullrelease script from the coredev buildout.

i18ndude is not needed either: i18ndude is also run from coredev to go over all packages to get the messages. Could still be useful for checking though.

Personally, I have both these tools installed globally as well.

**Review by mauritsvanrees — APPROVED** [review]
Thank you! The Bool/required change is very much needed yes.
I have added some comments, but they are really minor, so I approve.

File: setup.py (hunk 1 above)
Comment (1letter) [reply setup.py:46]: @mauritsvanrees if i remove the `"plone.app.robotframework[debug]",` then the tests are broken.

```
> bin/test

Module: plone.portlet.collection.tests.test_functional_doctest

Traceback (most recent call last):
  File "/opt/development/collective/plone.portlet.collection/plone/portlet/collection/tests/test_functional_doctest.py", line 6, in <module>
    from plone.portlet.collection.testing import PLONE_PORTLET_COLLECTION_FUNCTIONAL_TESTING
  File "/opt/development/collective/plone.portlet.collection/plone/portlet/collection/testing.py", line 2, in <module>
    from plone.app.contenttypes.testing import PLONE_APP_CONTENTTYPES_FIXTURE
  File "/home/1letter/.buildout/eggs/cp38/plone.app.contenttypes-3.0.0a11-py3.8.egg/plone/app/contenttypes/testing.py", line 4, in <module>
    from plone.app.event.testing import PAEvent_FIXTURE
  File "/home/1letter/.buildout/eggs/cp38/plone.app.event-4.0.0a8-py3.8.egg/plone/app/event/testing.py", line 3, in <module>
    from plone.app.robotframework.testing import AUTOLOGIN_LIBRARY_FIXTURE
ModuleNotFoundError: No module named 'plone.app.robotframework'


Module: plone.portlet.collection.tests.test_portlet_collection

Traceback (most recent call last):
  File "/opt/development/collective/plone.portlet.collection/plone/portlet/collection/tests/test_portlet_collection.py", line 6, in <module>
    from plone.portlet.collection.testing import (
  File "/opt/development/collective/plone.portlet.collection/plone/portlet/collection/testing.py", line 2, in <module>
    from plone.app.contenttypes.testing import PLONE_APP_CONTENTTYPES_FIXTURE
  File "/home/1letter/.buildout/eggs/cp38/plone.app.contenttypes-3.0.0a11-py3.8.egg/plone/app/contenttypes/testing.py", line 4, in <module>
    from plone.app.event.testing import PAEvent_FIXTURE
  File "/home/1letter/.buildout/eggs/cp38/plone.app.event-4.0.0a8-py3.8.egg/plone/app/event/testing.py", line 3, in <module>
    from plone.app.robotframework.testing import AUTOLOGIN_LIBRARY_FIXTURE
ModuleNotFoundError: No module named 'plone.app.robotframework'
```

File: base.cfg (hunk 2 above)
Comment (1letter) [reply base.cfg:86]: @mauritsvanrees ok, i will be remove this

File: base.cfg (hunk 3 above)
Comment (1letter) [reply base.cfg:105]: @mauritsvanrees ok, i will be remove this

File: base.cfg (hunk 4 above)
Comment (1letter) [reply base.cfg:14]: @mauritsvanrees ok, i will be remove this

File: setup.py (hunk 1 above)
Comment (mauritsvanrees) [reply setup.py:46]: Ah, that one again.
Two possible solutions.

- Add `plone.app.contenttypes[test]` to the test requirements. That pulls in the needed extra packages, including p.a.robotframework. Not ideal, but looks slightly nicer than explicitly including it ourselves.
- Better: do not use `PLONE_APP_CONTENTTYPES_FIXTURE` in `testing.py`, but use `from plone.app.testing import PLONE_FIXTURE`.

Comment (1letter) [reply setup.py]: @mauritsvanrees i have decided me for option two. Test are running.

**Comment by mauritsvanrees** [comment]
Merged. Thanks!

**Pull request merged**